components/Form/FormQuantitySpecification/index.js

```diff
@@ -0,0 +1,140 @@
+import styled from "styled-components";
+import SubmitButton from "../../Buttons/SubmitButton/SubmitButton";
+import ResetButton from "../../Buttons/ResetButton/ResetButton";
+import BasicButton from "../../Buttons/BasicButton/BasicButton";
+import Link from "next/link";
+import { useState } from "react";
+import {
+  DetailsScoreCalculationDeviation,
+  DetailsScoreCalculationAccordance,
+} from "../../../utils/DetailsScoreCalculation";
+
+export default function FormQuantitiySpecification({ foodCategory }) {
+  const [showResult, setShowResult] = useState(false);
+  const [resultText, setResultText] = useState("");
+  const [quantity, setQuantity] = useState(0);
+
+  const {
+    id,
+    name,
+    recommendedConsumption,
+    recommendedExample,
+    maxRange,
+    maxRangeInputField,
+  } = foodCategory;
+
+  function handleQuantitySpecification(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    setQuantity(parseFloat(data.consumedQuantity));
+    const consumedQuantityAccordance = DetailsScoreCalculationAccordance(
+      DetailsScoreCalculationDeviation,
+      quantity,
+      recommendedConsumption,
+      maxRange
+    );
+    const result = `Dein Konsum der Lebensmittelgruppe ${name} stimmt zu ${Math.floor(
+      consumedQuantityAccordance
+    )} % mit den Empfehlungen der Planetary Health Diet überein. Empfohlen werden ${recommendedConsumption} gramm. Das entspricht etwa: ${recommendedExample}.`;
+    setResultText(result);
+    setShowResult(true);
+    event.target.reset();
+  }
+
+  return (
+    <>
+      <Form
+        aria-labelledby="title"
+        aria-describedby="description"
+        onSubmit={handleQuantitySpecification}
+      >
+        <h3 id="title">Tägliches Essensquiz</h3>
+        <legend id="description">
+          Bitte wähle aus, wieviel <b>{name}</b> du heute gegessen hast:
+        </legend>
+        <StyledSection>
+          <div>0 g</div>
+          <input
+            step={1}
+            min={0}
+            type="range"
+            max={maxRangeInputField}
+            id={id}
+            name="consumedQuantity"
+            value={quantity}
+            quanitity={quantity}
+            onChange={(event) => {
+              setQuantity(event.target.value);
+            }}
+          />
+          <div>{`${maxRangeInputField} g`}</div>
+        </StyledSection>
+        <output>{`${quantity} gramm`}</output>
+        <ButtonBox>
+          <Link href="../..">
+            <BasicButton text="Zurück" />
+          </Link>
```

Comment:
The `href` is strange; and I would recommend to style the Link as Button instead of putting the BasicButton inside of the Link.